src/rantanplan/rhymes.py

```diff
@@ -0,0 +1,190 @@
+import re
+import statistics
+import string
+
+from spacy_affixes.utils import strip_accents
+
+
+def get_stressed_endings(scansion):
+    """Return a list of word endings, from the stressed position,
+    from a scansion list of tokens"""
+    endings = []
+    for line in scansion:
+        words = [word for word in line["tokens"] if "symbol" not in word]
+        syllables_count = sum(len(word["word"]) for word in words)
+        last_word = words[-1]
+        last_word_ending = last_word["word"][last_word["stress_position"]:]
+        ending = [syllable["syllable"] for syllable in last_word_ending]
+        endings.append((ending, syllables_count, last_word["stress_position"]))
+    return endings
+
+
+CONSONANTS = r"bcdfghjklmnñpqrstvwxyz"
+UNSTRESSED_VOWELS = r"aeiou"
+STRESSED_VOWELS = r"áéíóúäëïöü"
+WEAK_VOWELS = r"iuïü"
+STRONG_VOWELS = r"aeoáéó"
+WEAK_VOWELS_RE = re.compile(fr'[{WEAK_VOWELS}]([{STRONG_VOWELS}])',
+                            re.U | re.I)
+VOWELS = fr"{UNSTRESSED_VOWELS}{STRESSED_VOWELS}"
+STRESSED_VOWELS_RE = re.compile(fr'[{STRESSED_VOWELS}]', re.U | re.I)
+CONSONANTS_RE = re.compile(fr'[{CONSONANTS}]+', re.U | re.I)
+INITIAL_CONSONANTS_RE = re.compile(fr'^[{CONSONANTS}]+', re.U | re.I)
+DIPHTHONG_H_RE = re.compile(fr'([{VOWELS}])h([{VOWELS}])', re.U | re.I)
+DIPHTHONG_Y_RE = re.compile(fr'([{VOWELS}])h?y([^{VOWELS}])', re.U | re.I)
+STRUCTURES = (
+    (
+        "consonant",
+        "sonnet",
+        r"(abba|abab|cddc|cdcd){2}((cd|ef){3}|(cde|efg){2})",
+        lambda syllables: all(13 > syl > 10 for syl in syllables)
+    ),
+    ("consonant", "couplet", r"aa", lambda x: True),
+    (
+        "assonant",
+        "romance",
+        r"((.b)+)|((.a)+)",
+        lambda syllables: statistics.median(syllables) == 8
+    ),
+    (
+        "assonant",
```

Comment:
I think haikus don't have to rhyme. If I recall correctly the only rules are that they have to be 5-7-5 syllables and with a reference to a season (although the latter is way harder to implement). Is there an option to avoid rhyming detection? I think we should have one

Reply:
Hm. That's why the rhyme regular expression will match any rhyme pattern, `r".*"`. I'll add a comment to clarify what all those elements in the tuples do.